components/servo/webview.rs

```diff
@@ -2,34 +2,58 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
-use std::cell::RefCell;
+use std::cell::{Ref, RefCell, RefMut};
 use std::ffi::c_void;
-use std::rc::Rc;
+use std::hash::Hash;
+use std::rc::{Rc, Weak};
 use std::time::Duration;
 
 use base::id::WebViewId;
 use compositing::windowing::{MouseWindowEvent, WebRenderDebugOption};
 use compositing::IOCompositor;
 use compositing_traits::ConstellationMsg;
 use embedder_traits::{
-    ClipboardEventType, GamepadEvent, MediaSessionActionType, Theme, TouchEventType, TouchId,
-    TraversalDirection, WheelDelta,
+    ClipboardEventType, Cursor, GamepadEvent, LoadStatus, MediaSessionActionType, Theme,
+    TouchEventType, TouchId, TraversalDirection, WheelDelta,
 };
 use keyboard_types::{CompositionEvent, KeyboardEvent};
 use url::Url;
 use webrender_api::units::{DeviceIntPoint, DeviceIntSize, DevicePoint, DeviceRect};
 use webrender_api::ScrollLocation;
 
+use crate::webview_delegate::{DefaultWebViewDelegate, WebViewDelegate};
 use crate::ConstellationProxy;
 
 #[derive(Clone)]
-pub struct WebView(Rc<WebViewInner>);
+pub struct WebView(Rc<RefCell<WebViewInner>>);
```

Comment:
This doesn't need to change now, but I think WebView handle might need to cross the threads if the embedders have this demand. It will probably be `Arc<Mutex<T>>` in the future.

Reply:
Yes, we have discussed this. It would likely require a lot more locking. For now, embedders can do their own cross-thread communication to ensure that Servo stay on a single thread though.